maestro-ios-xctest-runner/maestro-driver-iosUITests/Routes/Handlers/LaunchAppHandler.swift

```diff
@@ -1,26 +1,51 @@
-import FlyingFox
+import Foundation
 import XCTest
+import FlyingFox
 import os
 
 @MainActor
 struct LaunchAppHandler: HTTPHandler {
     
     private let logger = Logger(
-        subsystem: Bundle.main.bundleIdentifier!,
-        category: String(describing: Self.self)
+           subsystem: Bundle.main.bundleIdentifier!,
+           category: String(describing: Self.self)
     )
- 
-    func handleRequest(_ request: FlyingFox.HTTPRequest) async throws -> HTTPResponse {
+    
+    func handleRequest(_ request: FlyingFox.HTTPRequest) async throws -> FlyingFox.HTTPResponse {
+        // Decode request body to extract appId and optional launch arguments
         guard let requestBody = try? await JSONDecoder().decode(LaunchAppRequest.self, from: request.bodyData) else {
-            return AppError(type: .precondition, message: "incorrect request body provided").httpResponse
+            return AppError(type: .precondition, message: "Incorrect request body for launching app").httpResponse
         }
         
-        NSLog("[Start] Launching app with bundle ID: \(requestBody.bundleId)")
-        XCUIApplication(bundleIdentifier: requestBody.bundleId).activate()
-        NSLog("[Done] Launching app with bundle ID: \(requestBody.bundleId)")
+        let app = XCUIApplication(bundleIdentifier: requestBody.appId)
 
-        
-        return HTTPResponse(statusCode: .ok)
+        // Set launch arguments if available
+        let arguments = requestBody.launchArguments
+        let flattenedArgs = arguments.flatMap { key, value -> [String] in
+            switch value.value {
+            case let bool as Bool:
+                return bool ? ["--\(key)"] : []
+            default:
+                return ["--\(key)", "\(value)"]
```

Comment:
I am not sure I got this flatmap logic? 

Why do we want to check the bool arguments separately here. Can't we just pass arguments as is?

Reply:
The reason we're using flatMap with a special case for Bool is because of how launchArguments work in XCUIApplication. It's not a dictionary but an array of strings, similar to command-line arguments.

For boolean flags:

If the value is true, we include it as a standalone flag like --featureFlagA.

If the value is false, we omit it entirely (which is the typical convention for CLI-like flags).

Reply:
Ummm, not sure I still got fully. For example, if we omit it and if someone is expecting false value on their app, will it give them nil or false?

I think right now your test also doesn't seem to cover this case. If you can ensure we have all the types covered in the test that would help here.